cow_instrument/DetectorInfo.h

```diff
@@ -40,27 +40,45 @@ template <typename InstTree> class DetectorInfo {
 
   bool isMonitor(size_t detectorIndex) const;
 
-  void initL2();
+  double l2(size_t detectorIndex) const;
 
-  void initL1();
+  Eigen::Vector3d position(size_t componentIndex) const;
 
-  double l2(size_t detectorIndex) const;
+  Eigen::Vector3d positionDetector(size_t detectorIndex) const;
```

Comment:
I'm with you on that. Accessing components via something claiming to be to do with `Detectors` is not very logical. I'd have to give this some thought, but one option would be to.

1. Refactor `ComponentInfo` (remove it in it's current form) and have just a `FlatTree`
2. Rename what is currently `DetectorInfo` to `ComponentInfo`
3. Put anything using detector indexes into a new class `DetectorInfo`. Make that use `ComponentInfo`.